File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        set<int> setBucket;
        int n=nums.size();
        for(int i=0;i<n;i++){
            setBucket.insert(nums[i]);
        }
        int result=0;
        for(int i=0;i<n;i++){
            int currentElement = nums[i];
            int previousElement = currentElement-1;
            int count=1;
            if(setBucket.find(previousElement)==setBucket.end()){
                while(setBucket.find(currentElement+1)!=setBucket.end()){
                    currentElement++;
                    count++;
                }
            }
            result=max(result, count);
        }
        return result;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int n=nums.length;
        int result=0;
        HashSet<Integer> set = new HashSet<>();
        for(int i=0;i<n;i++){
            set.add(nums[i]);
        }
        for(int i=0;i<n;i++){
            int currentElement=nums[i];
            int previousElement=currentElement-1;
            int count=1;
            if(!set.contains(previousElement)){
                while(set.contains(currentElement+1)){
                    currentElement++;
                    count++;
                }
            }
            result=Math.max(result, count);
        }
        return result;
    }
}